.devcontainer/setup.sh

```diff
@@ -0,0 +1,4 @@
+poetry install
```

Comment:
I don't know the Devcontainer workflow. The poetry run pre-commit install confuse me. I don't know if the purpose of setup.sh is run before start working in the new feature/bug/etc, in that way the pre-commit install make sense for me, but the other commands don't. 
Or when I finished to work in my new feature/bug/etc the setup.sh run, in that case I don't know if I want run the pre-commit install.
IMO it would be great write the new workflow (obviously if is necessary) in the documentation.

In any case, I like this idea.

Reply:
So, the `setup.sh` script runs just one time after the image has been created + the editor has launched (see postCreateCommand [here](https://containers.dev/implementors/json_reference/)).

The commands in this script are those which are denoted in https://python-poetry.org/docs/contributing/#local-development. You probably don't **need** to run `poetry run mypy` or `poetry run pytest` but I've included them to make sure everything is working as expected (doesn't really hurt to run them). 

This script/configuration doesn't change anything about the core development workflow as prescribed in the docs today.